packages/lib/server/service/insightsRouting.ts

```diff
@@ -60,101 +77,280 @@ export class InsightsRoutingService {
     this.filters = filters;
   }
 
-  async findMany(findManyArgs: Prisma.RoutingFormResponseDenormalizedFindManyArgs) {
+  /**
+   * Returns routing funnel data split by the given date ranges.
+   * @param dateRanges Array of { startDate, endDate, formattedDate }
+   */
+  async getRoutingFunnelData(dateRanges: DateRange[]) {
```

Comment:
I think this would be a prime place to cache data - even for 5 mins or something. Pretty intense query when i ran explain analysis